credentials/credentials.go

```diff
@@ -208,10 +205,23 @@ func (c *tlsCreds) OverrideServerName(serverNameOverride string) error {
 	return nil
 }
 
+const alpnProtoStrH2 = "h2"
+
+func appendH2ToNextProtos(ps []string) []string {
+	for _, p := range ps {
+		if p == alpnProtoStrH2 {
+			return ps
+		}
+	}
+	ret := make([]string, 0, len(ps)+1)
```

Comment:
How about:

```go
return append([]string{alpnProtoStrH2}, ps...)
```

Reply:
This is to append. I will check if the order matters.

Reply:
```go
        // NextProtos is a list of supported application level protocols, in
        // order of preference.
        NextProtos []string
```